### PR TITLE
way-shell: init package and module

### LINKED ISSUE #17: NixOS
Hi there,

Great project!

Is there a plan to package this for NixOS?

Best, 

M.

### LINKED ISSUE #329583: Package request: wayshell
**Project description**
A Gnome-like shell for wayland compositors

**Metadata**

* homepage URL: https://github.com/ldelossa/way-shell/
* source URL: https://github.com/ldelossa/way-shell/
* license: gpl2+
* platforms:  linux

---

I have so far done the following, but i just cannot get glib schema file to compile, i tried a lot. The binary itself compiles, but cannot run it as gschema is not available.

```nix
{ stdenv
, lib
, fetchFromGitHub
, pkg-config
, gtk-doc
, glib
, gtk4
, gsettings-desktop-schemas
, wrapGAppsHook4
, dconf
, libadwaita
, gtk4-layer-shell
, power-profiles-daemon
, upower
, wireplumber
, pipewire
, json-glib
, networkmanager
, libpulseaudio
, wayland-protocols
, wayland-utils
, xwayland
}:

stdenv.mkDerivation rec {
  pname = "wayshell";
  version = "0.1.0";

  src = fetchFromGitHub {
    owner = "ldelossa";
    repo = "way-shell";
    rev = "7c9d9c676914fc7f8df1ef10bf4fbfe70b4fd5bb";
    sha256 = "sha256-pT8e5t3ORlQcawcMhvcpWvBluPmK3tvTbyW/WdFlQPo=";
  };

  nativeBuildInputs = [
    pkg-config
    gtk-doc
    glib
    wrapGAppsHook4
  ];

  buildInputs = [
    libadwaita
    upower
    wireplumber
    json-glib
    networkmanager.dev
    libpulseaudio
    gtk4
    gtk4-layer-shell
    power-profiles-daemon
    gsettings-desktop-schemas
    dconf.lib
    pipewire
    pipewire.dev
    wayland-protocols
    wayland-utils
    xwayland
    glib.dev
  ];
  
  patches = [ ./add-pipewire-include.patch ];

  installPhase = ''
    make install DESTDIR=$out
    mkdir -p $out/share/glib-2.0/schemas
    glib-compile-schemas $out/share/glib-2.0/schemas
  '';

  #  postInstall = ''
  #    glib-compile-schemas $out/share/glib-2.0/schemas
  #  '';
  #make install-gschema DESTDIR=$out/share/glib-2.0/schemas

  # Apparently, No need to set SCHEMADIR explicitly as it likely defaults to share/glib-2.0/schemas

  preFixup = ''
    wrapProgram $out/usr/bin/way-shell \
      --prefix GIO_EXTRA_MODULES : "${dconf}/lib/gio/modules" \
      --prefix XDG_DATA_DIRS : "$out/share" \
      --prefix XDG_DATA_DIRS : "$out/share/gsettings-schemas/${gsettings-desktop-schemas.name}" \
      --prefix XDG_DATA_DIRS : "${gsettings-desktop-schemas}/share/gsettings-schemas/${gsettings-desktop-schemas.name}" \
      --prefix GSETTINGS_SCHEMA_DIR : "$out/share" \
      --prefix GSETTINGS_SCHEMAS_PATH : "$out/share/glib-2.0/schemas"
  '';

  meta = with lib; {
    homepage = "https://github.com/ldelossa/way-shell/";
    description = "A Gnome-like shell for wayland compositors";
    mainProgram = "way-shell";
    license = licenses.gpl2;
    platforms = platforms.unix;
  };
}
  ```
  
  patch file
  ```patch
  diff --git a/Makefile b/Makefile
index a3b6e7d..f7c2d12 100644
--- a/Makefile
+++ b/Makefile
@@ -11,7 +11,8 @@
 # depedency order matters here.
 # Gtk4LayerShell must be linked before any wayland libraries
 DEPS := libadwaita-1 \
-        gtk4-layer-shell-0 \
+        gtk4-layer-shell-0 \
+        libpipewire-0.3 \
        upower-glib \
        wireplumber-0.5 \
        json-glib-1.0 \
```

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by eclairevoyant**
## Description of changes

Adds the package and module for [way-shell](https://github.com/ldelossa/way-shell/), a GNOME-like shell for Wayland compositors.

- **way-shell: init at 0.0.7**
- **nixos/way-shell: init**

Fixes #329583
Fixes https://github.com/ldelossa/way-shell/issues/17

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [x] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/wa/way-shell/package.nix [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    pipewire
+    upower
+    wayland-protocols
+    wireplumber
+  ];
+
+  makeFlags = [ "PREFIX=$(out)" ];
+
+  postInstall = ''
+    make $makeFlags install-gschema
```
Comment (AndersonTorres) [inline pkgs/by-name/wa/way-shell/package.nix:59]: installFlags, maybe?

**Comment by ldelossa** [comment]
:wave: thanks for this! way-shell author here.

I'm not too familiar with NixOS, but I'm trying to understand, with the NixOS package essentially follow GitHub releases?

**Comment by eclairevoyant** [comment]
Switched to use the releases and added an update script. The update bot (r-ryantm) would pick up updates as long as the deps/build process hasn't changed; if it has I'd have to bump versions manually.

**Comment by sudanchapagain** [comment]
Why was this deleted? I assume deleting it, closed the PR automatically.

**Comment by NotAShelf** [comment]
I'll take this over.

**Comment by NotAShelf** [comment]
Continued in https://github.com/NixOS/nixpkgs/pull/343051, feel free to leave a review or any comments that you might have.

**Pull request closed without merging**